core/services/pipeline/orm.go

```diff
@@ -133,43 +132,17 @@ func (o *orm) CreateRun(ctx context.Context, jobID int32, meta map[string]interf
 			})
 		}
 		runID = run.ID
-		return tx.Create(&trs).Error
+		if len(trs) > 0 {
+			return tx.Create(&trs).Error
+		}
+		return nil
 	})
 	return runID, errors.WithStack(err)
 }
 
-// TODO: Remove generation of special "result" task
-// TODO: Remove the unique index on successor_id
-// https://www.pivotaltracker.com/story/show/176557536
 type ProcessRunFunc func(ctx context.Context, txdb *gorm.DB, spec Spec, l logger.Logger) (TaskRunResults, bool, error)
 
-// ProcessNextUnfinishedRun pulls the next available unfinished run from the
```

Comment:
Removing this complicated retry as there is a db poller anyways and this code isn't currently used

Reply:
We are definitely edging closer to removing it completely.

I'll wait til the direct request task is in before committing tho.